### ISSUE TITLE
Kindly help me with audioBook.Lcp Failure when come to the end of file.   

### ISSUE THREAD

**Issue by hishampro5** [readium/kotlin-toolkit#601]
### Describe the bug

i am developing android app for run  audiobooks.lcp ..etc but the issue only encounter with audiobooks (Lcp) when am trying same file on TestApp show me message that is not supported on test app.

The issue : 
when audio books run it's works fine but when audioBook finsh the error occurs you can see logs below i am not sure if this happen because readium or something else. 
 
 also there are design error when you click on drawer menu twice the back to book  the drawer menu items still appear above of book content. 
 

Error Log : 
UserError: An error occurred in the ExoPlayer engine. ExoPlayerEngine.Error.Engine: An error occurred in the ExoPlayer engine. ExoPlaybackException: Source error androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(ExoPlayerImplInternal.java:717) androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:693) Loader.UnexpectedLoaderException: Unexpected IllegalArgumentException: Failed requirement. androidx.media3.exoplayer.upstream.Loader$LoadTask.run(Loader.java:442) java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) IllegalArgumentException: Failed requirement. org.readium.r2.lcp.LcpDecryptor$CbcLcpResource.lengthFromLastTwoBlocks(LcpDecryptor.kt:150) org.readium.r2.lcp.LcpDecryptor$CbcLcpResource.read(LcpDecryptor.kt:236)

### How to reproduce?

1- open audioBook (LCP) 
2- play the book
3- wait to the end 
4- the error occurs 

### Readium version

3.0.2

### Android API version

14

**Comment by qnga** [comment]
These definitely look like Readium and testapp bugs, thanks for reporting. Your audiobook is probably guilty, but we should handle this case better. Could you confirm that you don't get this log with every LCP audiobook? I don't get it with a random one.

**Comment by hishampro5** [comment]
I'm sorry that this happens with all books of the same type ( Lcp ) 

also we tried on test app and get same issue Test app download from => https://github.com/readium/kotlin-toolkit/releases/download/3.0.2/test-app-3.0.2.apk

![Screenshot_٢٠٢٤-١٢-٣٠-١١-٢٢-٤٨-٩٨٧_org readium r2reader](https://github.com/user-attachments/assets/efe7e376-68e0-4ced-a289-d358e9b7124a)

![1735549673936](https://github.com/user-attachments/assets/f5c2a177-baea-47c4-a7af-94386a1afad3)


This is book you can try from your side (Lcp) book => 
Book Link on drive => https://drive.google.com/file/d/1jWTlh5fKTtK_zA5u8mBWmvw2JtR-8cl0/view?usp=sharing
Passphrase => malhadadi139gmailcommalhadadi139

@qnga

**Comment by qnga** [comment]
I can reproduce the bug, thanks again!

**Comment by hishampro5** [comment]
@qnga sorry sir but Any guess when this might be fixed

**Comment by qnga** [comment]
I'd like to review carefully this piece of code because it's not the first time it's faulty, maybe this week. In the meantime, you can use the `lcp-fixes` branch if you're in a hurry.

**Comment by hishampro5** [comment]
@qnga Hi sir is there any update for this issue.

**Comment by qnga** [comment]
I rewrote the code, expect a release in the next few days.

**Comment by hishampro5** [comment]
@mickael-menu  yes it's working fine with audio books (Lcp)  we tested with testApp . Can i know when this might be released

**Comment by qnga** [comment]
It's done!
